docs/auth/social-auth.md

```diff
@@ -327,26 +327,53 @@ with the new authentication state of the user.
+Per the [documentation](https://firebase.google.com/docs/auth/android/microsoft-oauth#expandable-1), we cannot handle the Sign-In flow manually, by getting the access token from a library such as `react-native-app-auth`, and then calling `signInWithCredential`.
+Instead, we must use the native's Sign-In flow from the Firebase SDK.
 
-```js
-import auth from '@react-native-firebase/auth';
-import { GoogleSignin } from '@react-native-google-signin/google-signin';
+To get started, please follow the prerequisites and setup instructions from the documentation: [Android](https://firebase.google.com/docs/auth/android/microsoft-oauth#before_you_begin), [iOS](https://firebase.google.com/docs/auth/ios/microsoft-oauth#before_you_begin).
 
-async function onGoogleLinkButtonPress() {
-  // Check if your device supports Google Play
-  await GoogleSignin.hasPlayServices({ showPlayServicesUpdateDialog: true });
-  // Get the user ID token
-  const { idToken } = await GoogleSignin.signIn();
+Additionally, for iOS, please follow step 1 of the "Handle sign-in flow" [section](https://firebase.google.com/docs/auth/ios/microsoft-oauth#handle_the_sign-in_flow_with_the_firebase_sdk), which is to add the custom URL scheme to your Xcode project
 
-  // Create a Google credential with the token
-  const googleCredential = auth.GoogleAuthProvider.credential(idToken);
+Once completed, setup your application to trigger a sign-in request with Microsoft using either of the `signInWithPopup` or `signInWithRedirect` methods. The underlying implementation is the same and will not operate exactly as the firebase-js-sdk web-based implementations do, but will provide drop-in compatibility for a web implementation if your project has one.
 
-  // Link the user with the credential
-  const firebaseUserCredential = await auth().currentUser.linkWithCredential(googleCredential);
-  // You can store in your app that the account was linked.
-  return;
+```jsx
+import React from 'react';
+import { Button } from 'react-native';
+
+function MicrosoftSignIn() {
+  return (
+    <Button
+      title="Microsoft Sign-In"
+      onPress={() => onMicrosoftButtonPress().then(() => console.log('Signed in with Microsoft!'))}
+    />
+  );
 }
 ```
+
+`onMicrosoftButtonPress` can be implemented as the following:
+
+```js
+import auth from '@react-native-firebase/auth';
+
+const onMicrosoftButtonPress = async () => {
+  // Generate the provider object
+  const provider = new auth.OAuthProvider('microsoft.com');
+  // Optionally add scopes
+  provider.addScope('offline_access');
+  // Optionally add custom parameters
+  provider.setCustomParameters({
+    prompt: 'consent',
+    tenant" 'tenant_name_or_id', //i.e. example.com or 9aaa9999-9999-999a-a9aa-9999aa9aa99a
```

Comment:
```suggestion
     // Optional "tenant" parameter for optional use of Azure AD tenant.
     // e.g., specific ID - 9aaa9999-9999-999a-a9aa-9999aa9aa99a or domain - example.com
     // defaults to "common" for tenant-independent tokens.
    tenant": 'tenant_name_or_id',
```


- Add missing colon here `(tenant" 'tenant_name_or_id')`
- Add better description of this critical field, because I lost like a lot of time because I didn't know that I need 'common' value in order to accept accounts that are not inside my tenant. It's just a suggestion.

Reply:
I like it! I do not have direct + personal experience with that, but reading your suggestion and comment, your suggestion looks great and it sounds like it will really help future devs. Thanks

Reply:
That's great. For my part, when i first tried it, i lost a  time figuring out how to restrict to a single tenant :)